optuna/integration/allennlp/_train.py

```diff
@@ -0,0 +1,438 @@
+"""
```

Comment:
In my understanding, most of this file is a transfer of AllenNLP code, and Optuna is only involved in the part that passes trial to `allennlp.commands.train.TrainModel`, restores `_CachedStorage`, and catches `TrialPruned`. As a suggestion, how about adding these Optuna-specific parts as an integration to the AllenNLP side? That is, put these changes together and send a PR to AllenNLP. Is this difficult to do?

Reply:
I created the issue on the AllenNLP repo (https://github.com/allenai/allennlp/issues/5338).
I'd like to go on this PR for now and migrate if the feature request is approved. What do you think about the plan?

Reply:
Sounds reasonable. Thank you for creating the issue in AllenNLP. I agree with your idea. Let's proceed with this PR.

Reply:
@toshihikoyanase Do you have any opinion on this? I'm very glad if you provide any insights.

Reply:
Sounds good!
After we merge this PR, we need to catch up with the changes of AllenNLP not to stale this integration.
But, I'm not sure if they include Optuna as a (mandatory) dependency, and I guess some plugin features may be preferable.
Anyway, thanks @himkt for creating the issue.